Added field to store number of random walk point sources

EiffL opened the pull request:
This pull request just adds a new field in `GalSimCelestialObject` to store the number of point sources in random walk light profiles instead of hijacking the sersic index

EiffL commented:
woups, I thought the tests were automatically run when opening a pull request, let me check that I didn't break anything

EiffL commented:
ok, passes all the tests:
```
========================================================================= 55 passed, 16 warnings in 502.03 seconds ==========================================================================
Global pytest run completed successfully
scons: Nothing to be done for `examples'.
scons: done building targets.
```
@danielsf while I'm at it, I could add a test for the RandomWalk (#41) . Do you have specific things that you would like to see checked ? Or a specific existing test you would want me to extend ?

danielsf commented:
This looks good as it is.

Regarding the unit test:

At the minimum, just having a unit test that draws a RandomWalk object so that we will know if we every break `drawRandomWalk()` would be sufficient.  I don't even think it would have to work like the other unit tests, which create a database and generate images from it in a CatSim-like way, since the CatSim database doesn't contain any RandomWalk galaxies.  You could just write a test that instantiates a `galSimInterpreter` by hand and calls `drawRandomWalk` directly.

If we want to be more quantitative, it might be nice to make sure that flux is actually conserved.  The test could draw an image with no background and only one RandomWalk galaxy, and then confirm that the total number of counts in the image is within a few sigma of what we expected.  That sort of test is done around this line:

https://github.com/lsst/sims_GalSimInterface/blob/master/tests/testAllowedChips.py#L202

I think that would be enough.

If you want to get this current branch merged, though, I would be happy making the unit test a part of a different PR.  Let me know.

EiffL commented:
@danielsf Thanks, I'm adding a test case to `testHalfLightRadius.py` then, but in the process of doing this I found some bugs, should be resolved in a couple of hours but **don't merge this branch just yet** .

EiffL commented:
@danielsf  Ok done, everything works and I have added a test in `testHalfLightRadius.py` with a random walk light profile. 
I have also added a travis CI script copied/pasted from the imSim repo, it executes all the tests but I have had to add a few lines to manually download the test data for the `afw` and  `sims_catUtils` packages. I don't know if there is a way to add the test data automatically when building a docker image for the DM stack...

In any case, this is ready to merge now [![Build Status](https://travis-ci.org/EiffL/sims_GalSimInterface.svg?branch=master)](https://travis-ci.org/EiffL/sims_GalSimInterface)

EiffL commented:
Also, this PR should address #39 and #41

danielsf commented on .travis.yml:
```diff
@@ -0,0 +1,29 @@
+script:
+  - sudo chown -R 1000:1000 .
+  - echo ${TRAVIS_BUILD_DIR}
+  - docker run -v `pwd`:${TRAVIS_BUILD_DIR} -e TRAVIS_BUILD_DIR lsstdesc/stack-sims:w_2018_09-sims_2_6_0-v2 ${TRAVIS_BUILD_DIR}/travis_test.sh
```
`master` of this repo needs to be buildable against `master` of the LSST data management repos.  Tagging the Travis build to a specific weekly could result in confusion when DM has explicitly made something not backwards compatible.  If Travis requires that the build be tagged to a specific version, then I would ask that we not use Travis. The LSST repos have their own Jenkins cluster against which we test before merging all pull requests.  I acknowledge that system is not helpful in a fork-and-pull-request world.  We will make sure to fix anything that is broken due to a merge from a fork.

EiffL replied:
ok thanks, I didn't know about that. Is the reason why imSim has such a travis script because it's a desc software instead of an lsst software?

danielsf replied:
Yes.  ImSim is DESC, rather than LSST.

ImSim has also adopted policies such that it only has to build against a specified tag of the lsst_sims stack.

Good idea, though.

danielsf commented on tests/testHalfLightRadius.py:
```diff
@@ -100,34 +119,36 @@ class characterizing the detector corresponding to this image
         yPixList = activePoints[0]  # arrays, the first index indicates what row it is in (the y coordinate)
         chipNameList = [detector.getName()]*len(xPixList)
 
+        # Compute luminosity weighted centroid
```
Thanks for fixing this.

EiffL commented:
@danielsf I have removed the travis scripts to avoid confusion :-) But I don't have merging powers so feel free to merge
